### PR TITLE
Update Dialog installs update

### PR DESCRIPTION
**Pull request by Naatan**
https://www.pivotaltracker.com/story/show/177824506

### CODE REVIEW

File: cmd/state-installer/installer.go [hunk 1]
```diff
@@ -111,11 +117,9 @@ func install(installPath string, cfg *config.Instance, out output.Outputer) erro
 	}
 
 	tmpDir := filepath.Dir(exe)
-	// clean-up temp directory when we are done.
-	defer os.RemoveAll(tmpDir)
-
 	inst := installer.New(filepath.Join(tmpDir, "bin"), installPath)
 	defer func() {
+		os.RemoveAll(tmpDir)
```
Comment (mdrohmann) [inline cmd/state-installer/installer.go:122]: Did you remove this for debugging purposes?

File: cmd/state-update-dialog/app.go [hunk 2]
```diff
@@ -48,30 +49,19 @@ func (a *App) CurrentVersion() string {
...
-	if update == nil {
-		return errs.New("No updates available")
-	}
-	bindings.update = update
-
-	cfg, err := config.New()
+	bindings := &Bindings{cfg: a.cfg}
+	var err error
+	// bindings.update, err = updater.NewChecker(constants.APIUpdateURL, "master", "", httpreq.New()).Check()
+	bindings.update, err = updater.DefaultChecker.Check() // TODO: Replace with this!
```
Comment (mdrohmann) [inline cmd/state-update-dialog/app.go:55]:
```suggestion
	bindings.update, err = updater.DefaultChecker.Check()
```

File: scripts/s3-deployer/main.go [hunk 3]
```diff
@@ -53,12 +53,25 @@ func run() {
...
+func (l *logger) Log(v ...interface{}) {
+	fmt.Printf("AWS Log: %v", v)
+}
+
 func createSession() {
 	// Specify profile to load for the session's config
 	var err error
+	var verboseErr = true
+	var logLevel = aws.LogDebug
+	_ = logLevel
```
Comment (mdrohmann) [inline scripts/s3-deployer/main.go:67]: Do we still need the verbose error reporting?

**Review by mdrohmann — CHANGES_REQUESTED** [review]

File: scripts/s3-deployer/main.go (hunk 3 above)
Comment (Naatan) [reply scripts/s3-deployer/main.go:67]: I left it in there so it's easy to toggle if we need it in the future.

File: cmd/state-installer/installer.go (hunk 1 above)
Comment (Naatan) [reply cmd/state-installer/installer.go:122]: I didn't? It's just in a single defer now.

Comment (mdrohmann) [reply cmd/state-installer/installer.go:122]: Oh dear! How silly of me...

**Review by mdrohmann — CHANGES_REQUESTED** [review]
I finally got to run it on Windows, tested it, and the Close button never gets enabled.  So, apparently, the `installDone` variable is never set to `true`...

**Review by mdrohmann — APPROVED** [review]

**Pull request merged**